src/CMakeLists.txt

```diff
@@ -302,7 +302,7 @@ macro(upm_swig_java)
     # See issue #518
     # TODO: Fix this somehow
     swig_add_module (javaupm_${libname} java javaupm_${libname}.i ${module_src})
-    swig_link_libraries (javaupm_${libname} ${MRAAJAVA_LIBRARIES} ${MRAA_LIBRARIES} ${JAVA_LIBRARIES} ${libname} interfaces)
+    swig_link_libraries (javaupm_${libname} ${MRAAJAVA_LIBRARY} ${JAVA_LIBRARIES} ${libname} interfaces)
     target_include_directories ( ${SWIG_MODULE_javaupm_${libname}_REAL_NAME}
```

Comment:
Hmm... I'm wondering if an MRAA dependency should just automatically include the MRAAJAVA dependency, as I think it's required if using Java and MRAA, but not required at all if your driver (and therefore Java example) is not using MRAA...?